### ISSUE TITLE
[Accessibility][Date Time Picker] The focus on the dateTime Picker is lost if the keyboard user clears the value

### ISSUE THREAD

**Issue by yonghuay** [carbon-design-system/carbon#8886]
## Environment

Mac OS

Chrome

## Detailed description

The focus on the DateTime Picker is lost if the keyboard user clears the values
 
> What version of the Carbon Design System are you using?

"carbon-components": "^10.33.0",

> What did you expect to happen?

The keyboard user can see the focus on the component even if s/he clears the input

> What happened instead?

The keyboard user **cannot** see the focus after s/he clears the input from the date time picker component.

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Step one - use the keyboard to select the start date and end date.
2. Step two - tab onto the start date to clear the input.
3. Step three - use the down arrow key to select the date again.

> Please create a reduced test case in CodeSandbox

You can view this issue on the carbon demo - https://www.carbondesignsystem.com/

## Additional information

- Screenshots or code
![Untitled 2](https://user-images.githubusercontent.com/32554015/121671600-a8c81980-caa6-11eb-9583-851349bbe2b6.gif)
